sleap/nn/data/instance_cropping.py

```diff
@@ -24,19 +25,24 @@ def find_instance_crop_size(
             architecture.
         input_scaling: Float factor indicating the scale of the input images if any
             scaling will be done before cropping.
+        crop_size: The (optional) crop size set by the user. None if not set.
 
     Returns:
         An integer crop size denoting the length of the side of the bounding boxes that
-        will contain the instances when cropped.
+        will contain the instances when cropped. The returned crop size will be larger
+        or equal to the input `crop_size`.
 
         This accounts for stride, padding and scaling when ensuring divisibility.
     """
+    crop_size = 0 if crop_size is None else crop_size
+
     max_length = 0.0
     for inst in labels.user_instances:
         pts = inst.points_array
         pts *= input_scaling
         max_length = np.maximum(max_length, np.nanmax(pts[:, 0]) - np.nanmin(pts[:, 0]))
         max_length = np.maximum(max_length, np.nanmax(pts[:, 1]) - np.nanmin(pts[:, 1]))
+        max_length = np.maximum(max_length, crop_size)
```

Comment:
I'm a little concerned about this since we add the padding afterwards.

Scenario:
1. `crop_size` was `None` in the first round of training, and this function computes it and returns `160`.
2. The user labels more and wants to retrain, so by default we pull in the config from the last round, which prepopulates that field with `160`.
3. It gets passed to this function again. Now it's `160` + `padding` = `176`.

Like, should we maybe just check if the `min_crop_size` is divisible already and/or assume it already includes padding?

Reply:
We now check if `min_crop_size` is divisible and also assume it already includes padding.